script/testing/junit/src/NestedQueryTest.java

```diff
@@ -0,0 +1,365 @@
+        stmt.execute(SQL_CREATE_SHIPMENT);
+        stmt.execute(SQL_CREATE_SUPPLIER);
+        stmt.execute(SQL_CREATE_PART);
+    }
+
+    /**
+     * Setup for each test, make the default connection
+     */
+    @Before
+    public void setup() throws SQLException {
+        try {
+            conn = makeDefaultConnection();
+            conn.setAutoCommit(true);
+            initDatabase();
+        } catch (SQLException e) {
+            DumpSQLException(e);
+        }
+    }
+
+    /**
+     * Cleanup for each test, execute after each test
+     * drop the default table and close connection
+     */
+    @After
+    public void teardown() throws SQLException {
+        try {
+            Statement stmt = conn.createStatement();
+            stmt.execute(SQL_DROP_SHIPMENT);
+            stmt.execute(SQL_DROP_SUPPLIER);
+            stmt.execute(SQL_DROP_PART);
+        } catch (SQLException e) {
+            DumpSQLException(e);
+        } finally {
+            try {
+                if (conn != null) {
+                    conn.close();
+                }
+            } catch (SQLException e) {
+                DumpSQLException(e);
+            }
+        }
+    }
+
+    /*
+     * ---------------------------------------------
+     * Nested query tests
+     * ---------------------------------------------
+     */
+    /**
+     * Test TypeA, max
+     */
+    @Test
+    public void test1TypeAMaxSimple() throws SQLException {
+        Statement stmt = conn.createStatement();
+        String sql = "INSERT INTO shipment VALUES (2, 2, 2);";
+        stmt.execute(sql);
+        sql = "INSERT INTO shipment VALUES (1, 1, 1);";
+        stmt.execute(sql);
+        sql = "INSERT INTO shipment VALUES (3, 3, 3);";
+        stmt.execute(sql);
+        sql = "INSERT INTO part VALUES (1, 30);";
+        stmt.execute(sql);
+        sql = "INSERT INTO part VALUES (2, 30);";
+        stmt.execute(sql);
+        String select_SQL = "SELECT sno FROM shipment WHERE pno = (SELECT MAX(pno) FROM part WHERE price = 30);";
+        rs = stmt.executeQuery(select_SQL);
+        rs.next();
+        checkIntRow(rs, new String [] {"sno"}, new int [] {2});
+        assertNoMoreRows(rs);
+    }
+
+    /**
+     * Test TypeA, max, multiple results
+     */
+    @Test
+    public void test2TypeAMaxMulti() throws SQLException {
+        Statement stmt = conn.createStatement();
+        String sql = "INSERT INTO shipment VALUES (2, 2, 2);";
+        stmt.execute(sql);
+        sql = "INSERT INTO shipment VALUES (1, 1, 1);";
+        stmt.execute(sql);
+        sql = "INSERT INTO shipment VALUES (3, 2, 3);";
+        stmt.execute(sql);
+        sql = "INSERT INTO part VALUES (1, 30);";
+        stmt.execute(sql);
+        sql = "INSERT INTO part VALUES (2, 30);";
+        stmt.execute(sql);
+        String select_SQL = "SELECT sno FROM shipment WHERE pno = (SELECT MAX(pno) FROM part WHERE price = 30);";
+        rs = stmt.executeQuery(select_SQL);
+        rs.next();
```

Comment:
Again please check no of columns returned

Reply:
added